src/NuGet.Core/NuGet.Packaging/PackageExtractor.cs

```diff
@@ -1088,7 +1088,7 @@ private static void DeleteTargetAndTempPaths(string targetPath, string targetTem
                        token,
                        parentId);
 
-                if (verifyResult.Signed)
+                if (verifyResult.Signed || !verifyResult.Valid)
```

Comment:
This would create a bad error for the situations in which the result is invalid. 

I'm not sure if this is a likely scenario today, but the checks can still be improved.

Reply:
Why do you say this will have a bad error? Line 1093 prints out a generic verbose message to announce it has verified the package. The message is `PackageSignatureVerificationLog: PackageIdentity: {0} Source: {1} PackageSignatureValidity: {2}.` ... Then lines 1105-1118, do the appropriate behavior for valid/invalid verification. All the behavior seems to be what is intended for either signed packages that needed to be verified, or unsigned packages that are not permitted to be unsigned (therefore invalid verification).

Reply:
Discussed offline, I misunderstood where this codepath was being called.